### PR TITLE
1.1.5 Changes

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: registration/registration-services/src/main/java/io/mosip/registration/util/acktemplate/TemplateGenerator.java [hunk 1]
```diff
@@ -575,4 +614,169 @@ private String getImage(String imagePath) {
...
+		}
+		List<Map<String, Object>> updateList = new ArrayList<>();
+		updateList.add(appInstalledTime);
+		updateList.add(lastSWUpdate);
+		activities.put(RegistrationConstants.DASHBOARD_UPDATES, updateList);
+		return activities;
+	}
+	
+	private String trimJobName(String jobName) {		
+		return jobName.substring(0, jobName.lastIndexOf(" "));
```
Comment (ghost) [inline registration/registration-services/src/main/java/io/mosip/registration/util/acktemplate/TemplateGenerator.java]: why are we not calling trim() directly ?

File: registration/registration-client/src/main/java/io/mosip/registration/controller/auth/LoginController.java [hunk 2]
```diff
@@ -842,51 +846,69 @@ public void loadLoginScreen(String loginMode) {
...
-			if (Desktop.isDesktopSupported()) {
-				try {
-					Desktop.getDesktop().browse(new URI(RegistrationConstants.MOSIP_URL));
-				} catch (IOException ioException) {
-					LOGGER.error(LoggerConstants.LOG_REG_LOGIN, APPLICATION_NAME, APPLICATION_ID,
-							ioException.getMessage() + ExceptionUtils.getStackTrace(ioException));
-				} catch (URISyntaxException uriSyntaxException) {
-					LOGGER.error(LoggerConstants.LOG_REG_LOGIN, APPLICATION_NAME, APPLICATION_ID,
-							uriSyntaxException.getMessage() + ExceptionUtils.getStackTrace(uriSyntaxException));
+			if (RegistrationAppHealthCheckUtil.isNetworkAvailable()) {
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-client/src/main/java/io/mosip/registration/controller/auth/LoginController.java]: Removed the check as suggested.

File: registration/registration-client/src/main/java/io/mosip/registration/controller/auth/LoginController.java [hunk 3]
```diff
@@ -842,51 +846,69 @@ public void loadLoginScreen(String loginMode) {
...
-					Desktop.getDesktop().browse(new URI(RegistrationConstants.MOSIP_URL));
-				} catch (IOException ioException) {
-					LOGGER.error(LoggerConstants.LOG_REG_LOGIN, APPLICATION_NAME, APPLICATION_ID,
-							ioException.getMessage() + ExceptionUtils.getStackTrace(ioException));
-				} catch (URISyntaxException uriSyntaxException) {
-					LOGGER.error(LoggerConstants.LOG_REG_LOGIN, APPLICATION_NAME, APPLICATION_ID,
-							uriSyntaxException.getMessage() + ExceptionUtils.getStackTrace(uriSyntaxException));
+	public void forgotPword(ActionEvent event) {
+		forgotPword.setOnAction(e -> {
+			if (RegistrationAppHealthCheckUtil.isNetworkAvailable()) {
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-client/src/main/java/io/mosip/registration/controller/auth/LoginController.java]: Removed the check for internet connection.

File: registration/registration-client/src/main/java/io/mosip/registration/controller/reg/HeaderController.java [hunk 4]
```diff
@@ -889,6 +892,34 @@ public void userGuide(ActionEvent event) {
...
 		});
 	}
+	
+	/**
+	 * Redirects to mosip.io in case of user reset pword
+	 * 
+	 * @param event event for reset pword
+	 */
+	public void resetPword(ActionEvent event) {
+		if (RegistrationAppHealthCheckUtil.isNetworkAvailable()) {
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-client/src/main/java/io/mosip/registration/controller/reg/HeaderController.java]: Removed the check for internet connection.

File: registration/registration-client/src/main/java/io/mosip/registration/controller/reg/PacketHandlerController.java [hunk 5]
```diff
@@ -1034,6 +1024,54 @@ public void loadReRegistrationScreen() {
...
+			LOGGER.error("REGISTRATION - LOAD_DASHBOARD_SCREEN - REGISTRATION_OFFICER_PACKET_CONTROLLER",
+					APPLICATION_NAME, APPLICATION_ID, exception.getMessage() + ExceptionUtils.getStackTrace(exception));
+
+			generateAlert(RegistrationConstants.ERROR, RegistrationUIConstants.UNABLE_LOAD_DASHBOARD_PAGE);
+		}
+		LOGGER.info(PACKET_HANDLER, APPLICATION_NAME, APPLICATION_ID, "Loading dashboard screen ended.");
+	}
+
+	// TODO Remove this method
+	private String getDashBoardTemplate() {
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-client/src/main/java/io/mosip/registration/controller/reg/PacketHandlerController.java]: Yes. Added it for testing purpose. Forgot to remove that piece of code. Removed now.

File: registration/registration-services/src/main/java/io/mosip/registration/constants/RegistrationConstants.java [hunk 6]
```diff
@@ -1833,4 +1839,33 @@ private RegistrationConstants() {
...
\ No newline at end of file
+  
+  public static final String FORGOT_USERNAME_URL = "mosip.registration.forgot_username_url";
+	public static final String FORGOT_PWORD_URL = "mosip.registration.forgot_password_url";
+	public static final String RESET_PWORD_URL = "mosip.registration.reset_password_url";
+	public static final String EMAIL_PLACEHOLDER = "{EMAIL}";
+
+	// Dashboard Constants
+	public static final String DASHBOARD_TITLE = "dashBoard";
+	public static final String DASHBOARD_FORMAT = "dd MMM hh:mm a";
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/constants/RegistrationConstants.java]: Modified it and added the format to spring.properties file. Fetching it from there.

File: registration/registration-services/src/main/java/io/mosip/registration/entity/SyncJobDef.java [hunk 7]
```diff
@@ -38,6 +38,8 @@
 	private Boolean isDeleted;
 	@Column(name = "del_dtimes")
 	private Timestamp deletedDateTime;
+	@Column(name = "job_type")
+	private String jobType;
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/entity/SyncJobDef.java:42]: Added rollback scripts to drop the column.

File: registration/registration-services/src/main/java/io/mosip/registration/service/bio/impl/BioServiceImpl.java [hunk 8]
```diff
@@ -130,10 +130,18 @@ protected RegistrationDTO getRegistrationDTOFromSession() {
 								.getBioProvider(biometricType, BiometricFunction.QUALITY_CHECK)
 								.getModalityQuality(birList, null);
 
-						LOGGER.info(BIO_SERVICE, APPLICATION_NAME, APPLICATION_ID,
-								"Quality score is evaluated and assigning to biometricsDto..");
-
-						biometricsDto.setQualityScore(scoreMap.get(biometricType));
+						if (ApplicationContext.map().containsKey(RegistrationConstants.UPDATE_SDK_QUALITY_SCORE)
```
[the remark this answers is not in the document]
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/service/bio/impl/BioServiceImpl.java]: Used getOrDefault method to get the value.

File: registration/registration-services/src/main/java/io/mosip/registration/util/acktemplate/TemplateGenerator.java (hunk 1 above)
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/util/acktemplate/TemplateGenerator.java]: Added this method just to show the purpose clearly. Removed and called trim() directly as it was called only once.

**Pull request merged**